source/Mlos.Python/mlos/Spaces/Hypergrid.py

```diff
@@ -50,13 +59,73 @@ def get_dimensions_for_point(self, point, return_join_dimensions=True):
     def random(self, point=None):
         raise NotImplementedError("All subclasses must implement this.")
 
+    @trace()
     def random_dataframe(self, num_samples):
         config_dicts = [
             {dim_name: value for dim_name, value in self.random()}
             for _ in range(num_samples)
         ]
         return pd.DataFrame(config_dicts)
 
+    @trace()
+    def filter_out_invalid_rows(self, original_dataframe: pd.DataFrame, exclude_extra_columns=True) -> pd.DataFrame:
+        """ Returns a dataframe containing only valid rows from the dataframe.
+
+        Valid rows are rows with no NaNs and with values for all dimensions in the required ranges.
+
+        If there are additional columns, they will be preserved.
+
+        :param original_dataframe:

```

Comment:
Do you want to use the numpydoc standard? Then it'll render properly on the sphinx website.

Reply:
Yeah, I'll do that from now on.

Reply:
And I'll fix this PR.

Reply:
Thanks! I think it would be great to move to it bit-by-bit whenever we work on docs.

Reply:
Once those are more cleaned up we can also consider to add that as part of CI checks